Fix Error JSON_DATA_TEXT in souffle-compile.py on Apple Silicon

trganda opened the pull request:
## Detail of the error on m1/m2 serials mac os.

While install with the command on mac os with applie silicon chips.

```
brew install --HEAD souffle-lang/souffle/souffle
```

The generated souffle-compile.py file has an invalid JSON_DATA_TEXT configuration as show below.

```
#!/usr/bin/env python3
JSON_DATA_TEXT = """{
  "compiler": "/opt/homebrew/Library/Homebrew/shims/mac/super/g++-11",
  "compiler_id": "AppleClang",
  "compiler_version": "14.0.0.14000029",
  "msvc_version": "",
  "includes": "-I/tmp/souffle-20221019-29532-1c32bqh/src/include -I/Library/Frameworks/Mono.framework/Headers",
  "std_flag": "-std=c++17",
  "cxx_flags": "",
  "cxx_link_flags": "-Wl,-search_paths_first -Wl,-headerpad_max_install_names",
  "release_cxx_flags": "-O3 ",
  "debug_cxx_flags": "-g",
  "definitions": "-DUSE_NCURSES -DUSE_LIBZ -DUSE_SQLITE",
  "compile_options": "",
  "link_options": "/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib/libsqlite3.tbd /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib/libz.tbd /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib/libncurses.tbd",
  "rpaths": "/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib:/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib:/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib",
  "outname_fmt": "-o {}",
  "libdir_fmt": "-L{}",
  "libname_fmt": "-l{}",
  "rpath_fmt": "-Wl,-rpath,{}",
  "path_delimiter": ":",
  "exe_extension": "",
  "source_include_dir": "/tmp/souffle-20221019-29532-1c32bqh/src/include",
  "jni_includes": ":"
}"""

# --- JSON_DATA_TEXT variable is inserted before this line ---
```

There has two problem, 
* the `/opt/homebrew/Library/Homebrew/shims/mac/super/g++-11` of homebrew cannot be use directly, otherwise an error info [g++-11: The build tool has reset ENV; --env=std required.](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/shims/super/cc) will be throws.
* the include directory `/tmp/souffle-20221019-29532-1c32bqh/src/include` was a temporary directory, it will be deleted after installation.

Theres problem will leading to the `souffle -o` failed, because compile failed. For more detail of this issue see [souffle-lang/souffle/issues/2325](https://github.com/souffle-lang/souffle/issues/2325).

trganda commented:
I have test this formual on my machine. 

![WX20221022-104800@2x](https://user-images.githubusercontent.com/62204882/197315820-0bb11f2a-2d7c-4666-89a9-df694b6bf79e.png)


The uname info

```
Darwin loaclhost 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:19:52 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T6000 arm64
```

The detail of installation with `brew install -v --HEAD souffle-lang/souffle/souffle`

```
➜  vendor git:(master) brew install -v --HEAD souffle-lang/souffle/souffle
Running `brew update --auto-update`...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).

You have 4 outdated formulae installed.
You can upgrade them with brew upgrade
or list them with brew outdated.

==> Cloning https://github.com/souffle-lang/souffle.git
Updating /Users/trganda/Library/Caches/Homebrew/souffle--git
git config remote.origin.url https://github.com/souffle-lang/souffle.git
git config remote.origin.fetch \+refs/heads/master:refs/remotes/origin/master
git config remote.origin.tagOpt --no-tags
git config advice.detachedHead false
git fetch origin
==> Checking out branch master
git checkout -f master --
Already on 'master'
Your branch is up to date with 'origin/master'.
git reset --hard origin/master --
HEAD is now at 29c5921e6 Merge pull request #2331 from quentin/osxci
==> Installing souffle from souffle-lang/souffle
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/man/. /private/tmp/d20221022-21265-1rvnsej/man
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/CMakeLists.txt /private/tmp/d20221022-21265-1rvnsej/CMakeLists.txt
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/.codecov.yml /private/tmp/d20221022-21265-1rvnsej/.codecov.yml
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/LICENSE /private/tmp/d20221022-21265-1rvnsej/LICENSE
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/debian/. /private/tmp/d20221022-21265-1rvnsej/debian
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/cmake/. /private/tmp/d20221022-21265-1rvnsej/cmake
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/licenses/. /private/tmp/d20221022-21265-1rvnsej/licenses
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/.clang-tidy /private/tmp/d20221022-21265-1rvnsej/.clang-tidy
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/tests/. /private/tmp/d20221022-21265-1rvnsej/tests
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/README.md /private/tmp/d20221022-21265-1rvnsej/README.md
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/sh/. /private/tmp/d20221022-21265-1rvnsej/sh
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/utilities/. /private/tmp/d20221022-21265-1rvnsej/utilities
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/.gitignore /private/tmp/d20221022-21265-1rvnsej/.gitignore
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/.clang-format /private/tmp/d20221022-21265-1rvnsej/.clang-format
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/.github/. /private/tmp/d20221022-21265-1rvnsej/.github
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/.gitattributes /private/tmp/d20221022-21265-1rvnsej/.gitattributes
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/choco-packages.config /private/tmp/d20221022-21265-1rvnsej/choco-packages.config
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/.git/. /private/tmp/d20221022-21265-1rvnsej/.git
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/doxygen.in /private/tmp/d20221022-21265-1rvnsej/doxygen.in
cp -pR /Users/trganda/Library/Caches/Homebrew/souffle--git/src/. /private/tmp/d20221022-21265-1rvnsej/src
cp -pR /private/tmp/d20221022-21265-1rvnsej/man/. /private/tmp/souffle-20221022-21265-1iu1lpo/man
cp -pR /private/tmp/d20221022-21265-1rvnsej/CMakeLists.txt /private/tmp/souffle-20221022-21265-1iu1lpo/CMakeLists.txt
cp -pR /private/tmp/d20221022-21265-1rvnsej/.codecov.yml /private/tmp/souffle-20221022-21265-1iu1lpo/.codecov.yml
cp -pR /private/tmp/d20221022-21265-1rvnsej/LICENSE /private/tmp/souffle-20221022-21265-1iu1lpo/LICENSE
cp -pR /private/tmp/d20221022-21265-1rvnsej/debian/. /private/tmp/souffle-20221022-21265-1iu1lpo/debian
cp -pR /private/tmp/d20221022-21265-1rvnsej/cmake/. /private/tmp/souffle-20221022-21265-1iu1lpo/cmake
cp -pR /private/tmp/d20221022-21265-1rvnsej/licenses/. /private/tmp/souffle-20221022-21265-1iu1lpo/licenses
cp -pR /private/tmp/d20221022-21265-1rvnsej/.clang-tidy /private/tmp/souffle-20221022-21265-1iu1lpo/.clang-tidy
cp -pR /private/tmp/d20221022-21265-1rvnsej/tests/. /private/tmp/souffle-20221022-21265-1iu1lpo/tests
cp -pR /private/tmp/d20221022-21265-1rvnsej/README.md /private/tmp/souffle-20221022-21265-1iu1lpo/README.md
cp -pR /private/tmp/d20221022-21265-1rvnsej/sh/. /private/tmp/souffle-20221022-21265-1iu1lpo/sh
cp -pR /private/tmp/d20221022-21265-1rvnsej/utilities/. /private/tmp/souffle-20221022-21265-1iu1lpo/utilities
cp -pR /private/tmp/d20221022-21265-1rvnsej/.gitignore /private/tmp/souffle-20221022-21265-1iu1lpo/.gitignore
cp -pR /private/tmp/d20221022-21265-1rvnsej/.clang-format /private/tmp/souffle-20221022-21265-1iu1lpo/.clang-format
cp -pR /private/tmp/d20221022-21265-1rvnsej/.github/. /private/tmp/souffle-20221022-21265-1iu1lpo/.github
cp -pR /private/tmp/d20221022-21265-1rvnsej/.gitattributes /private/tmp/souffle-20221022-21265-1iu1lpo/.gitattributes
cp -pR /private/tmp/d20221022-21265-1rvnsej/choco-packages.config /private/tmp/souffle-20221022-21265-1iu1lpo/choco-packages.config
cp -pR /private/tmp/d20221022-21265-1rvnsej/.git/. /private/tmp/souffle-20221022-21265-1iu1lpo/.git
cp -pR /private/tmp/d20221022-21265-1rvnsej/doxygen.in /private/tmp/souffle-20221022-21265-1iu1lpo/doxygen.in
cp -pR /private/tmp/d20221022-21265-1rvnsej/src/. /private/tmp/souffle-20221022-21265-1iu1lpo/src
chmod -Rf +w /private/tmp/d20221022-21265-1rvnsej
==> git fetch --tags
==> cmake -B build -S . -DCMAKE_INSTALL_PREFIX=/opt/homebrew/Cellar/souffle/HEAD-29c5921 -DSOUFFLE_GIT=OFF -DSOUFFLE_BASH_COMPLETION=OFF -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++
-- The CXX compiler identification is AppleClang 14.0.0.14000029
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/homebrew/Library/Homebrew/shims/mac/super/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Build Type: Release
-- Found Curses: /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib/libcurses.tbd  
-- Found FLEX: /usr/bin/flex (found version "2.6.4") 
-- Found BISON: /opt/homebrew/opt/bison/bin/bison (found suitable version "3.8.2", minimum required is "3.0.4") 
-- Found ZLIB: /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib/libz.tbd (found version "1.2.11") 
-- Found SQLite3: /Library/Frameworks/Mono.framework/Headers (found version "3.26.0") 
-- Found LibFFI: /opt/homebrew/opt/libffi/include  
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
-- Could NOT find OpenMP (missing: OpenMP_CXX_FOUND) 
-- Found Python3: /opt/homebrew/Frameworks/Python.framework/Versions/3.10/bin/python3.10 (found suitable version "3.10.8", minimum required is "3.7") found components: Interpreter 
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_COMPILER


-- Build files have been written to: /tmp/souffle-20221022-21265-1iu1lpo/build
==> cmake -B build -S . -DCMAKE_INSTALL_PREFIX=/opt/homebrew/Cellar/souffle/HEAD-29c5921 -DSOUFFLE_GIT=OFF -DSOUFFLE_BASH_COMPLETION=OFF -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_CXX_FLAGS=-I/opt/homebrew/Cellar/souffle/HEAD-29c5921/include
-- Build Type: Release
-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
-- Could NOT find OpenMP (missing: OpenMP_CXX_FOUND) 
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/souffle-20221022-21265-1iu1lpo/build
==> cmake --build build --target install -j17
[  0%] [FLEX][scanner] Building scanner with flex 2.6.4
[  1%] [BISON][parser] Building parser with bison 3.8.2
[  1%] Building CXX object src/tests/CMakeFiles/test_brie.dir/brie_test.cpp.o
[  2%] Building CXX object src/CMakeFiles/compiled.dir/dummy.cpp.o
[  2%] Building CXX object src/CMakeFiles/souffleprof.dir/souffle_prof.cpp.o
[  3%] Building CXX object src/tests/CMakeFiles/test_btree_set.dir/btree_set_test.cpp.o
[  4%] Building CXX object src/tests/CMakeFiles/test_parallel_utils.dir/parallel_utils_test.cpp.o
[  4%] Building CXX object src/tests/CMakeFiles/test_disjoint_set_property.dir/disjoint_set_property_test.cpp.o
[  4%] Building CXX object src/tests/CMakeFiles/test_record_table.dir/record_table_test.cpp.o
[  5%] Building CXX object src/tests/CMakeFiles/test_flyweight.dir/flyweight_test.cpp.o
[  6%] Building CXX object src/tests/CMakeFiles/test_btree_multiset.dir/btree_multiset_test.cpp.o
[  6%] Building CXX object src/tests/CMakeFiles/test_profile_util.dir/profile_util_test.cpp.o
[  7%] Building CXX object src/tests/CMakeFiles/test_symbol_table.dir/symbol_table_test.cpp.o
[  7%] Building CXX object src/tests/CMakeFiles/test_compiled_tuple.dir/compiled_tuple_test.cpp.o
[  7%] Building CXX object src/tests/CMakeFiles/test_binary_relation.dir/binary_relation_test.cpp.o
[  7%] Building CXX object src/tests/CMakeFiles/test_graph_utils.dir/graph_utils_test.cpp.o
[  8%] Building CXX object src/tests/CMakeFiles/test_eqrel_datastructure.dir/eqrel_datastructure_test.cpp.o
[  8%] Linking CXX static library libcompiled.a
[  9%] Building CXX object src/tests/CMakeFiles/test_table.dir/table_test.cpp.o
warning: /Library/Developer/CommandLineTools/usr/bin/ranlib: archive library: libcompiled.a the table of contents is empty (no object file members in the library define global symbols)
[  9%] Built target compiled
[  9%] Building CXX object src/tests/CMakeFiles/test_util.dir/util_test.cpp.o
[  9%] Building CXX object src/CMakeFiles/libsouffle.dir/FunctorOps.cpp.o
[  9%] Linking CXX executable test_parallel_utils
[  9%] Built target test_parallel_utils
[ 10%] Building CXX object src/CMakeFiles/libsouffle.dir/Global.cpp.o
[ 10%] Linking CXX executable test_table
[ 11%] Linking CXX executable test_compiled_tuple
[ 11%] Built target test_table
[ 11%] Built target test_compiled_tuple
[ 11%] Building CXX object src/tests/CMakeFiles/test_visitor.dir/visitor_test.cpp.o
[ 12%] Building CXX object src/tests/CMakeFiles/test_getopt_long.dir/getopt_long_test.cpp.o
[ 12%] Linking CXX executable test_disjoint_set_property
[ 13%] Linking CXX executable test_graph_utils
[ 13%] Built target test_disjoint_set_property
[ 13%] Linking CXX executable test_symbol_table
[ 13%] Linking CXX executable test_flyweight
[ 13%] Built target test_graph_utils
[ 14%] Building CXX object tests/semantic/pragma2/CMakeFiles/lib_foo.dir/lib_foo.cpp.o
[ 14%] Building CXX object tests/semantic/pragma2/CMakeFiles/lib_bar.dir/lib_bar.cpp.o
[ 14%] Built target test_symbol_table
[ 14%] Built target test_flyweight
[ 14%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Aggregator.cpp.o
[ 15%] Building CXX object tests/semantic/functor_fact/CMakeFiles/functor_fact.dir/functors.cpp.o
[ 15%] Linking CXX shared library "lib'never gonna'.dylib"
[ 15%] Linking CXX shared library "lib'let you'.dylib"
[ 16%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/IntrinsicAggregator.cpp.o
[ 16%] Built target lib_foo
[ 16%] Built target lib_bar
[ 16%] Building CXX object tests/interface/aggregates/CMakeFiles/afunctors.dir/functors.cpp.o
[ 16%] Building CXX object tests/interface/functors/CMakeFiles/functors.dir/functors.cpp.o
[ 16%] Linking CXX shared library libfunctors.dylib
[ 16%] Built target functor_fact
[ 16%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/UserDefinedAggregator.cpp.o
[ 17%] Linking CXX executable test_profile_util
[ 18%] Linking CXX executable test_util
[ 18%] Building CXX object tests/interface/graph_coloring/CMakeFiles/gcfunctors.dir/functors.cpp.o
[ 18%] Built target test_profile_util
[ 18%] Built target test_util
[ 18%] Building CXX object tests/interface/pathseq/CMakeFiles/psfunctors.dir/functors.cpp.o
[ 19%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/AlgebraicDataType.cpp.o
[ 19%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/AliasType.cpp.o
[ 19%] Linking CXX shared library libfunctors.dylib
[ 19%] Built target afunctors
[ 19%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Atom.cpp.o
[ 20%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Attribute.cpp.o
[ 21%] Linking CXX shared library libfunctors.dylib
[ 21%] Built target gcfunctors
[ 21%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/BinaryConstraint.cpp.o
[ 22%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/BooleanConstraint.cpp.o
[ 23%] Linking CXX shared library libfunctors.dylib
[ 23%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/BranchType.cpp.o
[ 24%] Linking CXX shared library libfunctors.dylib
[ 24%] Built target psfunctors
[ 24%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/BranchInit.cpp.o
[ 24%] Linking CXX executable test_eqrel_datastructure
[ 24%] Linking CXX executable test_btree_multiset
[ 24%] Built target functors
[ 24%] Built target test_eqrel_datastructure
[ 25%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Clause.cpp.o
[ 25%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/SubsumptiveClause.cpp.o
[ 25%] Built target test_btree_multiset
[ 26%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Component.cpp.o
[ 26%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/ComponentInit.cpp.o
[ 27%] Linking CXX executable test_binary_relation
[ 27%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/ComponentType.cpp.o
[ 29%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Constant.cpp.o
[ 29%] Linking CXX executable test_visitor
[ 29%] Built target test_binary_relation
[ 29%] Linking CXX executable test_getopt_long
[ 29%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Counter.cpp.o
[ 29%] Built target test_visitor
[ 30%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Directive.cpp.o
[ 30%] Built target test_getopt_long
[ 30%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/ExecutionOrder.cpp.o
[ 30%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/ExecutionPlan.cpp.o
[ 31%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/FunctorDeclaration.cpp.o
[ 31%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/FunctionalConstraint.cpp.o
[ 32%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/IntrinsicFunctor.cpp.o
[ 32%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Negation.cpp.o
[ 33%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/NilConstant.cpp.o
[ 33%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Node.cpp.o
[ 33%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/NumericConstant.cpp.o
[ 33%] Linking CXX executable test_btree_set
[ 34%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Pragma.cpp.o
[ 34%] Built target test_btree_set
[ 34%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Program.cpp.o
[ 35%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/QualifiedName.cpp.o
[ 35%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/RecordInit.cpp.o
[ 35%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/RecordType.cpp.o
[ 36%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Relation.cpp.o
[ 36%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/StringConstant.cpp.o
[ 37%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/SubsetType.cpp.o
[ 37%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Term.cpp.o
[ 37%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/TranslationUnit.cpp.o
[ 38%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Type.cpp.o
[ 38%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/TypeCast.cpp.o
[ 39%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/UnionType.cpp.o
[ 39%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/UnnamedVariable.cpp.o
[ 40%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/UserDefinedFunctor.cpp.o
[ 40%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/Variable.cpp.o
[ 40%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/Aggregate.cpp.o
[ 41%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/ClauseNormalisation.cpp.o
[ 41%] Linking CXX executable test_record_table
[ 41%] Built target test_record_table
[ 41%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/ComponentLookup.cpp.o
[ 42%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/Functor.cpp.o
[ 42%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/Ground.cpp.o
[ 42%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/IOType.cpp.o
[ 43%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/PrecedenceGraph.cpp.o
[ 43%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/ProfileUse.cpp.o
[ 44%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/RecursiveClauses.cpp.o
[ 44%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/RedundantRelations.cpp.o
[ 44%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/RelationSchedule.cpp.o
[ 45%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/JoinSize.cpp.o
[ 45%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/SCCGraph.cpp.o
[ 46%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/TopologicallySortedSCCGraph.cpp.o
[ 46%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/typesystem/PolymorphicObjects.cpp.o
[ 46%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/typesystem/SumTypeBranches.cpp.o
[ 47%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/typesystem/Type.cpp.o
[ 47%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/typesystem/TypeConstraints.cpp.o
[ 48%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/typesystem/TypeConstrainsAnalysis.cpp.o
[ 48%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/typesystem/TypeSystem.cpp.o
[ 49%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/analysis/typesystem/TypeEnvironment.cpp.o
[ 49%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/AddNullariesToAtomlessAggregates.cpp.o
[ 49%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/ComponentChecker.cpp.o
[ 50%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/ComponentInstantiation.cpp.o
[ 50%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/DebugReporter.cpp.o
[ 51%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/ExecutionPlanChecker.cpp.o
[ 51%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/ExpandEqrels.cpp.o
[ 51%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/FoldAnonymousRecords.cpp.o
[ 52%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/GroundedTermsChecker.cpp.o
[ 52%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/GroundWitnesses.cpp.o
[ 53%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/InlineRelations.cpp.o
[ 53%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/MagicSet.cpp.o
[ 53%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/MaterializeAggregationQueries.cpp.o
[ 54%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/MaterializeSingletonAggregation.cpp.o
[ 54%] Linking CXX executable souffleprof
[ 54%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/Meta.cpp.o
[ 55%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/MinimiseProgram.cpp.o
[ 55%] Linking CXX executable test_brie
[ 55%] Built target souffleprof
[ 55%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/NameUnnamedVariables.cpp.o
[ 55%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/NormaliseGenerators.cpp.o
[ 55%] Built target test_brie
[ 56%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/PartitionBodyLiterals.cpp.o
[ 56%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/PragmaChecker.cpp.o
[ 57%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/ReduceExistentials.cpp.o
[ 57%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/RemoveBooleanConstraints.cpp.o
[ 58%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/RemoveEmptyRelations.cpp.o
[ 58%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/RemoveRedundantRelations.cpp.o
[ 58%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/RemoveRedundantSums.cpp.o
[ 59%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/RemoveRelationCopies.cpp.o
[ 59%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/ReplaceSingletonVariables.cpp.o
[ 60%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/ResolveAliases.cpp.o
[ 60%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/ResolveAnonymousRecordAliases.cpp.o
[ 60%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/SemanticChecker.cpp.o
[ 61%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/SubsumptionQualifier.cpp.o
[ 61%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/SimplifyAggregateTargetExpression.cpp.o
[ 62%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/Transformer.cpp.o
[ 62%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/TypeChecker.cpp.o
[ 62%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/transform/UniqueAggregationVariables.cpp.o
[ 63%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/utility/BindingStore.cpp.o
[ 63%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/utility/SipsMetric.cpp.o
[ 64%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/utility/SipGraph.cpp.o
[ 64%] Building CXX object src/CMakeFiles/libsouffle.dir/ast/utility/Utils.cpp.o
[ 65%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/provenance/ClauseTranslator.cpp.o
[ 65%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/provenance/ConstraintTranslator.cpp.o
[ 65%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/provenance/SubproofGenerator.cpp.o
[ 66%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/provenance/TranslationStrategy.cpp.o
[ 66%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/provenance/UnitTranslator.cpp.o
[ 67%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/seminaive/ClauseTranslator.cpp.o
[ 67%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/seminaive/ConstraintTranslator.cpp.o
[ 67%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/seminaive/TranslationStrategy.cpp.o
[ 68%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/seminaive/UnitTranslator.cpp.o
[ 68%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/seminaive/ValueTranslator.cpp.o
[ 69%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/utility/Utils.cpp.o
[ 69%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/utility/TranslatorContext.cpp.o
[ 69%] Building CXX object src/CMakeFiles/libsouffle.dir/ast2ram/utility/ValueIndex.cpp.o
[ 70%] Building CXX object src/CMakeFiles/libsouffle.dir/interpreter/Engine.cpp.o
[ 70%] Building CXX object src/CMakeFiles/libsouffle.dir/interpreter/Generator.cpp.o
[ 71%] Building CXX object src/CMakeFiles/libsouffle.dir/interpreter/BrieIndex.cpp.o
[ 71%] Building CXX object src/CMakeFiles/libsouffle.dir/interpreter/BTreeIndex.cpp.o
[ 71%] Building CXX object src/CMakeFiles/libsouffle.dir/interpreter/BTreeDeleteIndex.cpp.o
[ 72%] Building CXX object src/CMakeFiles/libsouffle.dir/interpreter/EqrelIndex.cpp.o
[ 72%] Building CXX object src/CMakeFiles/libsouffle.dir/interpreter/ProvenanceIndex.cpp.o
[ 73%] Building CXX object src/CMakeFiles/libsouffle.dir/parser/ParserDriver.cpp.o
[ 73%] Building CXX object src/CMakeFiles/libsouffle.dir/parser/ParserUtils.cpp.o
[ 74%] Building CXX object src/CMakeFiles/libsouffle.dir/parser/SrcLocation.cpp.o
[ 74%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/Node.cpp.o
[ 74%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/TranslationUnit.cpp.o
[ 75%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/analysis/Complexity.cpp.o
[ 75%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/analysis/Index.cpp.o
[ 76%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/analysis/Level.cpp.o
[ 76%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/analysis/Relation.cpp.o
[ 76%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/IfExistsConversion.cpp.o
[ 77%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/CollapseFilters.cpp.o
[ 77%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/EliminateDuplicates.cpp.o
[ 78%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/ExpandFilter.cpp.o
[ 78%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/HoistAggregate.cpp.o
[ 78%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/HoistConditions.cpp.o
[ 79%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/IfConversion.cpp.o
[ 79%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/MakeIndex.cpp.o
[ 80%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/Parallel.cpp.o
[ 80%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/ReorderConditions.cpp.o
[ 80%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/ReorderFilterBreak.cpp.o
[ 81%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/Transformer.cpp.o
[ 81%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/transform/TupleId.cpp.o
[ 82%] Building CXX object src/CMakeFiles/libsouffle.dir/ram/utility/NodeMapper.cpp.o
[ 82%] Building CXX object src/CMakeFiles/libsouffle.dir/reports/DebugReport.cpp.o
[ 83%] Building CXX object src/CMakeFiles/libsouffle.dir/synthesiser/Synthesiser.cpp.o
[ 83%] Building CXX object src/CMakeFiles/libsouffle.dir/synthesiser/Relation.cpp.o
[ 83%] Building CXX object src/CMakeFiles/libsouffle.dir/synthesiser/Utils.cpp.o
[ 84%] Building CXX object src/CMakeFiles/libsouffle.dir/synthesiser/GenDb.cpp.o
[ 84%] Building CXX object src/CMakeFiles/libsouffle.dir/parser/scanner.cc.o
[ 85%] Building CXX object src/CMakeFiles/libsouffle.dir/parser/parser.cc.o
[ 85%] Linking CXX static library libsouffle.a
[ 85%] Built target libsouffle
[ 85%] Building CXX object src/ast/tests/CMakeFiles/test_constraints.dir/constraints_test.cpp.o
[ 85%] Building CXX object src/ast/tests/CMakeFiles/test_type_system.dir/type_system_test.cpp.o
[ 86%] Building CXX object src/ast/tests/CMakeFiles/test_ast_transformers.dir/ast_transformers_test.cpp.o
[ 87%] Building CXX object src/ram/tests/CMakeFiles/test_max_matching.dir/max_matching_test.cpp.o
[ 88%] Building CXX object src/ram/tests/CMakeFiles/test_ram_relation_equal_clone.dir/ram_relation_equal_clone_test.cpp.o
[ 89%] Building CXX object src/ram/tests/CMakeFiles/test_ram_condition_equal_clone.dir/ram_condition_equal_clone_test.cpp.o
[ 89%] Building CXX object src/ast/tests/CMakeFiles/test_ast_print.dir/ast_print_test.cpp.o
[ 90%] Building CXX object src/ast/tests/CMakeFiles/test_ast_program.dir/ast_program_test.cpp.o
[ 90%] Building CXX object src/interpreter/tests/CMakeFiles/test_ram_arithmetic.dir/ram_arithmetic_test.cpp.o
[ 90%] Building CXX object src/ram/tests/CMakeFiles/test_ram_statement_equal_clone.dir/ram_statement_equal_clone_test.cpp.o
[ 90%] Building CXX object src/interpreter/tests/CMakeFiles/test_interpreter_relation.dir/interpreter_relation_test.cpp.o
[ 91%] Building CXX object src/interpreter/tests/CMakeFiles/test_ram_relation.dir/ram_relation_test.cpp.o
[ 91%] Building CXX object src/CMakeFiles/souffle.dir/main.cpp.o
[ 91%] Building CXX object src/ast/tests/CMakeFiles/test_ast_utils.dir/ast_utils_test.cpp.o
[ 91%] Building CXX object src/ram/tests/CMakeFiles/test_ram_type_conversion.dir/ram_type_conversion_test.cpp.o
[ 91%] Building CXX object src/ram/tests/CMakeFiles/test_matching.dir/matching_test.cpp.o
[ 92%] Building CXX object src/ram/tests/CMakeFiles/test_ram_expression_equal_clone.dir/ram_expression_equal_clone_test.cpp.o
[ 93%] Linking CXX executable test_ram_type_conversion
[ 93%] Linking CXX executable test_ram_relation_equal_clone
[ 93%] Built target test_ram_type_conversion
[ 93%] Built target test_ram_relation_equal_clone
[ 94%] Linking CXX executable test_constraints
[ 94%] Linking CXX executable test_max_matching
[ 94%] Linking CXX executable test_ast_print
[ 94%] Linking CXX executable test_matching
[ 95%] Linking CXX executable test_ast_utils
[ 95%] Built target test_constraints
[ 95%] Built target test_max_matching
[ 95%] Built target test_ast_print
[ 95%] Built target test_matching
[ 95%] Built target test_ast_utils
[ 95%] Linking CXX executable test_ast_program
[ 95%] Linking CXX executable test_ram_expression_equal_clone
[ 95%] Built target test_ast_program
[ 95%] Built target test_ram_expression_equal_clone
[ 96%] Linking CXX executable test_type_system
[ 96%] Linking CXX executable test_ram_condition_equal_clone
[ 96%] Built target test_ram_condition_equal_clone
[ 96%] Built target test_type_system
[ 96%] Linking CXX executable test_ast_transformers
[ 96%] Built target test_ast_transformers
[ 97%] Linking CXX executable test_ram_statement_equal_clone
[ 97%] Built target test_ram_statement_equal_clone
[ 98%] Linking CXX executable test_interpreter_relation
[ 98%] Built target test_interpreter_relation
[ 99%] Linking CXX executable test_ram_arithmetic
[ 99%] Built target test_ram_arithmetic
[ 99%] Linking CXX executable test_ram_relation
[ 99%] Built target test_ram_relation
[100%] Linking CXX executable souffle
[100%] Built target souffle
Install the project...
-- Install configuration: "Release"
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/TypeAttribute.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/swig
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/swig/SwigInterface.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/swig/SwigInterface.i
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io/ReadStreamCSV.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io/WriteStreamSQLite.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io/WriteStreamJSON.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io/WriteStreamCSV.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io/IOSystem.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io/ReadStreamSQLite.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io/gzfstream.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io/SerialisationStream.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io/ReadStreamJSON.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io/WriteStream.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/io/ReadStream.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/RecordTable.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/SymbolTable.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/SouffleFunctor.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/ConcurrentInsertOnlyHashMap.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/EqRel.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/SymbolTableImpl.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/Brie.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/ConcurrentFlyweight.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/BTreeUtil.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/LambdaBTree.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/BTreeDelete.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/RecordTableImpl.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/Graph.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/UnionFind.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/BTree.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/Table.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/PiggyList.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/Info.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/Nullaries.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/datastructure/EquivalenceRelation.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/provenance
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/provenance/ExplainProvenanceImpl.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/provenance/Explain.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/provenance/ExplainTree.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/provenance/ExplainProvenance.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/htmlMain.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/Reader.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/Relation.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/htmlJsChartistPlugin.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/DataComparator.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/Iteration.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/htmlJsTableSort.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/EventProcessor.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/htmlJsUtil.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/StringUtils.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/Cell.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/HtmlGenerator.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/ProgramRun.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/Cli.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/ProfileDatabase.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/CellInterface.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/htmlCssStyle.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/Tui.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/htmlCssChartist.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/Rule.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/ProfileEvent.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/Table.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/OutputProcessor.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/Logger.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/UserInputReader.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/htmlJsMain.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/Row.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/profile/htmlJsChartistMin.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/CompiledSouffle.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/SignalHandler.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/RamTypes.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/SouffleInterface.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/BinaryConstraintOps.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/SubsetCache.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/NodeMapper.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/tinyformat.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/GetOptLong.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/NodeMapperFwd.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/Iteration.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/Types.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/VisitorFwd.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/General.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/FileUtil.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/ParallelUtil.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/GetOptLongImpl.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/StringUtil.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/CacheUtil.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/Visitor.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/json11.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/ContainerUtil.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/MiscUtil.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/span.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/EvaluatorUtil.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/FunctionalUtil.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/SubProcess.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/StreamUtil.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/utility/DynamicCasting.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/include/souffle/CompiledOptions.h
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/bin/souffle
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/bin/souffleprof
-- Installing: /opt/homebrew/Cellar/souffle/HEAD-29c5921/bin/souffle-compile.py
==> sed -i .bak s#/opt/homebrew/Library/Homebrew/shims/mac/super/g++#/usr/bin/g++#g /opt/homebrew/Cellar/souffle/HEAD-29c5921/bin/souffle-compile.py
==> Cleaning
==> Finishing up
ln -s ../Cellar/souffle/HEAD-29c5921/bin/souffle souffle
ln -s ../Cellar/souffle/HEAD-29c5921/bin/souffle-compile.py souffle-compile.py
ln -s ../Cellar/souffle/HEAD-29c5921/bin/souffle-compile.py.bak souffle-compile.py.bak
ln -s ../Cellar/souffle/HEAD-29c5921/bin/souffleprof souffleprof
ln -s ../Cellar/souffle/HEAD-29c5921/include/souffle souffle
==> Summary
🍺  /opt/homebrew/Cellar/souffle/HEAD-29c5921: 104 files, 15.8MB, built in 1 minute 56 seconds
==> Running `brew cleanup souffle`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
```

b-scholz reviewed (approved):
Thanks for fixing it!

b-scholz commented:
Please, can you resolve the conflict and check whether it is still an issue? Many thanks in advance!

trganda commented:
> Please, can you resolve the conflict and check whether it is still an issue? Many thanks in advance!

Ok, I will check.

trganda commented:
Hi @b-scholz, I have resolve the conflicts, and added a building command `system "rm", "-rf", "build/"` to remove the cache before the new cmake build command.
And I have re-tested on my machine with [Command Line Tools for Xcode 14(September 12, 2022)](https://developer.apple.com/download/all/).

If there is no other problem, I think you can merge this commit.

```bash
➜  datalog brew uninstall souffle                      
Uninstalling /opt/homebrew/Cellar/souffle/2.3... (100 files, 11.1MB)
➜  datalog brew install --HEAD souffle-lang/souffle/souffle
==> Cloning https://github.com/souffle-lang/souffle.git
Updating /Users/trganda/Library/Caches/Homebrew/souffle--git
==> Checking out branch master
Previous HEAD position was cc8ea0917 Merge pull request #2272 from b-scholz/changelog2.3
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 8acdf4960 Merge pull request #2352 from adamjseitz/ignored-execution-order-errors
==> Installing souffle from souffle-lang/souffle
==> git fetch --tags
==> cmake -B build -S . -DSOUFFLE_BASH_COMPLETION=OFF -DCMAKE_CXX_COMPILER=g++-11
==> rm -rf build/
==> cmake -B build -S . -DSOUFFLE_BASH_COMPLETION=OFF -DCMAKE_CXX_COMPILER=g++ -DCMAKE_CXX_FLAGS=-I/opt/homebrew/Cellar/
==> cmake --build build --target install -j
==> sed -i .bak s#/opt/homebrew/Library/Homebrew/shims/mac/super/g++#/usr/bin/g++#g /opt/homebrew/Cellar/souffle/HEAD-8a
🍺  /opt/homebrew/Cellar/souffle/HEAD-8acdf49: 105 files, 15.9MB, built in 3 minutes 1 second
==> Running `brew cleanup souffle`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
➜  datalog ls
Magic.csv      alias.csv      example        example.dl     fib.csv        pointsTo.csv   test.py        varpointsto.dl
Translate.csv  edge.facts     example.cpp    example.log    path.csv       rule_demo1.dl  type_demo1.dl
➜  datalog souffle -F. -D. -oexample example.dl
➜  datalog ./example      
➜  datalog cat path.csv                  
1	2
1	3
2	3
```

b-scholz reviewed (approved):
Thanks for the fix!
